src/scripts/main.js

```diff
@@ -1,3 +1,223 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const [tHead, tBody] = table.children;
+const tableBodyRows = [...tBody.rows];
+const tablesTitles = [...tHead.children[0].cells]
+  .map(td => td.textContent.trim().toLowerCase());
+
+function sortedTable(index, typesSort, methodSort) {
+  const bodyRows = [...tBody.rows];
+
+  bodyRows.sort((a, b) => {
+    let partA = a.children[index].innerHTML;
+    let partB = b.children[index].innerHTML;
+
+    if (typesSort) {
+      partA = partA.replace(/\D/g, '');
+      partB = partB.replace(/\D/g, '');
+
+      return methodSort ? partA - partB : partB - partA;
+    }
+
+    return methodSort ? partA.localeCompare(partB) : partB.localeCompare(partA);
+  }).forEach(el => tBody.append(el));
+}
+
+tHead.addEventListener('click', (e) => {
+  const index = Array.from(e.target.parentElement.children)
+    .findIndex(el => el.innerHTML === e.target.innerHTML);
+  const wasSortingAlready = e.target.classList.contains('sort-asc');
+
+  if (wasSortingAlready) {
+    e.target.classList.remove('sort-asc');
+    e.target.classList.toggle('sort-desc');
+  } else {
+    tHead.querySelectorAll('th').forEach(th =>
+      th.classList.remove('sort-asc', 'sort-desc'));
+    e.target.classList.add('sort-asc');
+  }
+
+  const methodSort = e.target.classList.value === 'sort-asc' ? 1 : 0;
+  const typesSort = index <= 2 ? 0 : 1;
```

Comment:
```suggestion
  const methodSort = e.target.classList.value === 'sort-asc' ? 1 : 0;
  const typesSort = index <= 2 ? 0 : 1;
```
Було б круто придумати цим значенням якісь більш змістовні імена, а не 1 та 2